content/kb/troubleshooting-failure-to-delete-orphaned-pod-volume-directory.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Troubleshooting: Failure to delete orphaned pod volume directory"
+author: Ray Chang
+draft: false
+date: 2023-03-23
+categories:
+  - "csi"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+When a worker node crashes with some pods running on it, many error messages as shown below keep appearing in the kubelet.
+
+```
+orphaned pod <pod-uid> found, but error not a directory occurred when trying to remove the volumes dir
+```
+
+## Reason
+
+This is caused by the kubelet trying to tear down the volume by sending a CSI request to the longhorn-csi-plugin, but a node failure resulted in a disconnect. Then, when the kubelet tries to delete `vol_data.json` to unmount the volume, it also failed due to the failure of the CSI request.
+
+## Solution
+
+After the node and kubelet are restored, the longhorn-csi-plugin will automatically restart, and the pod with the new volume mounted will be at Running state again.
+
+However, if the replica is scheduled to another node and produces a dangling `vol_data.json` file on the crashed node already, then you need to manually delete `vol_data.json` which is under `/var/lib/kubelet/pods/<pod-uid>/volumes/kubernetes.io~csi/pvc_<pod-uid>/` directory.
+
+## Related information
+
+* Related Longhorn issue: https://github.com/longhorn/longhorn/issues/3207
+* Related Kubernetes issue: https://github.com/kubernetes/kubernetes/issues/105536
```

Comment:
Kubernetes PR https://github.com/kubernetes/kubernetes/pull/116134 supposedly fixes the tracked issue for Kubernetes v1.27.0+, though the tracked issue hasn't been verified or closed. Can we mention that we only expect this issue to occur in Kubernetes <v.1.27 somewhere in this KB?

Reply:
Thanks, I also found out that this is asked to backported to 1.27 but havn't got merge yet (https://github.com/kubernetes/kubernetes/pull/117235). But I think I could update this PR to include relevant information.